### ISSUE TITLE
Remove calls to deprecated jQuery functions

### ISSUE THREAD

**Issue by JohnJVTK** [kartik-v/bootstrap-fileinput#1856]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest `master` branch of bootstrap-fileinput.
- [X] This is not an usage question. I confirm having read the plugin [documentation](http://plugins.krajee.com/file-input) and [demos](http://plugins.krajee.com/file-input/demo).
- [X] This is not a general programming / coding question. (Those should be directed to the [webtips Q & A forum](http://webtips.krajee.com/questions)).
- [X] I have attempted to find the simplest possible steps to reproduce the issue.
- [ ] I have included a failing test as a pull request (Optional).

## Steps to reproduce the issue

Create a project with jQuery 3.7.1 and add jQuery Migrate 3.4.2-pre that will trigger warnings
![Capture d'écran 2024-02-29 094234](https://github.com/kartik-v/bootstrap-fileinput/assets/60505932/58df2c11-751f-4263-9b72-91f3b74677f9)

## Expected behavior and actual behavior

As mentionned in their [blog post](https://blog.jquery.com/2024/02/06/jquery-4-0-0-beta/), jQuery 4.0 will remove some functions still in use in bootstrap-fileinput 5.5.2 but deprecated for several versions of jQuery 3.x.

I found the following functions used by bootstrap-fileinput :
- [jQuery.isArray](https://github.com/jquery/jquery/pull/4364) deprecated since jQuery 3.2
- [jQuery.isFunction](https://github.com/jquery/jquery/pull/4364) deprecated since jQuery 3.3
- [jQuery.now](https://github.com/jquery/jquery/pull/4364) deprecated since jQuery 3.3
- [jQuery.isNumeric](https://github.com/jquery/jquery/pull/4364) deprecated since jQuery 3.3
- [jQuery.trim](https://github.com/jquery/jquery/pull/4461) deprecated since jQuery 3.5

I found the following code in [jQuery 3.7.1](https://code.jquery.com/jquery-3.7.1.js) but not present in 4.0.0 beta version.

```
jQuery.isArray = Array.isArray;

jQuery.isFunction = function( obj ) {

		// Support: Chrome <=57, Firefox <=52
		// In some browsers, typeof returns "function" for HTML <object> elements
		// (i.e., `typeof document.createElement( "object" ) === "function"`).
		// We don't want to classify *any* DOM node as a function.
		// Support: QtWeb <=3.8.5, WebKit <=534.34, wkhtmltopdf tool <=0.12.5
		// Plus for old WebKit, typeof returns "function" for HTML collections
		// (e.g., `typeof document.getElementsByTagName("div") === "function"`). (gh-4756)
		return typeof obj === "function" && typeof obj.nodeType !== "number" &&
			typeof obj.item !== "function";
	};

jQuery.now = Date.now;

jQuery.toType = function( obj ) {
	if ( obj == null ) {
		return obj + "";
	}

	// Support: Android <=2.3 only (functionish RegExp)
	return typeof obj === "object" || typeof obj === "function" ?
		class2type[ toString.call( obj ) ] || "object" :
		typeof obj;
}

jQuery.isNumeric = function( obj ) {

	// As of jQuery 3.0, isNumeric is limited to
	// strings and numbers (primitives or objects)
	// that can be coerced to finite numbers (gh-2662)
	var type = jQuery.type( obj );
	return ( type === "number" || type === "string" ) &&

		// parseFloat NaNs numeric-cast false positives ("")
		// ...but misinterprets leading-number strings, particularly hex literals ("0x...")
		// subtraction forces infinities to NaN
		!isNaN( obj - parseFloat( obj ) );
};

jQuery.trim = function( text ) {
	return text == null ?
		"" :
		( text + "" ).replace( rtrim, "$1" );
};
```

Deprecated functions should be replaced with their native equivalent otherwise it will break for users using jQuery 4.0.

## Environment

Browsers

- [X] Google Chrome 122
- [ ] Mozilla Firefox
- [ ] Internet Explorer
- [ ] Safari

Operating System

- [X] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 3.7.1 / 4.0.0 beta
- jQuery Migrate: 3.4.2-pre
- bootstrap-fileinput version: 5.5.2

## Isolating the problem

- [ ] This bug happens [on the plugin demos page](http://plugins.krajee.com/file-input/demo)
- [ ] The bug happens consistently across all tested browsers
- [ ] This bug happens when using bootstrap-fileinput without other plugins
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by kartik-v** [comment]
Thanks - can you submit a PR for a potential enhancement on this?

**Comment by JohnJVTK** [comment]
Honestly I don't have time to do this. There is a lot of changes and I prefer that you do it because I don't want to mess with your code.
I invite you to read [jQuery deprecated items](https://api.jquery.com/category/deprecated/) because jQuery Migrate plugin warned of others deprecated items like :
- jQuery.fn.removeAttr no longer sets boolean properties: disabled
- jQuery.fn.focus() event shorthand is deprecated
- ...

Moreover I used your plugin for years but I dropped it months ago when I removed jQuery and every project using it.
I just wanted to warn you and let you know that jQuery 4.0 will break your project.
